src/languageservice/parser/jsonParser07.ts

```diff
@@ -734,9 +735,13 @@ function validate(
         validate(node, subSchema, schema, subValidationResult, subMatchingSchemas, options);
         if (!subValidationResult.hasProblems() || callFromAutoComplete) {
           matches.push(subSchema);
+          subMatches.push(subSchema);
           if (subValidationResult.propertiesMatches === 0) {
             noPropertyMatches.push(subSchema);
           }
+          if (subSchema.format) {
```

Comment:
For address https://github.com/redhat-developer/yaml-language-server/issues/586 omit the matches when string format matched